pygmt/tests/test_clib_put_strings.py

```diff
@@ -0,0 +1,67 @@
+"""
+Test the functions that put string data into GMT.
+"""
+import numpy as np
+import numpy.testing as npt
+import pytest
+from packaging.version import Version
+
+from .. import clib
+from ..exceptions import GMTCLibError
+from ..helpers import GMTTempFile
+
+with clib.Session() as _lib:
+    gmt_version = Version(_lib.info["version"])
+
+
+@pytest.mark.xfail(
+    condition=gmt_version < Version("6.1.1"),
+    reason="GMT_Put_Strings only works for GMT 6.1.1 and above",
+)
+def test_put_strings():
+    "Check that assigning a numpy array of dtype str to a dataset works"
+    with clib.Session() as lib:
+        dataset = lib.create_data(
+            family="GMT_IS_DATASET|GMT_VIA_VECTOR",
+            geometry="GMT_IS_POINT",
+            mode="GMT_CONTAINER_ONLY",
+            dim=[2, 5, 1, 0],  # columns, rows, layers, dtype
+        )
+        x = np.array([1, 2, 3, 4, 5], dtype=np.int32)
+        y = np.array([6, 7, 8, 9, 10], dtype=np.int32)
+        strings = np.array(["a", "b", "c", "d", "e"], dtype=np.str)
```

Comment:
Please change the elements of the string array to different length, to test passing strings of variable lengths. For example:
```suggestion
        strings = np.array(["a", "bc", "defg", "hijklmn", "opqrst"], dtype=np.str)
```